TheJumpCloud/support
# CUT-4128: Missing Documentation Pester Tests

## Issues
* [CUT-4128](https://jumpcloud.atlassian.net/browse/CUT-4128) - Missing Documentation Pester Tests

## What does this solve?
Introduces a new Pester test to the `HelpFiles.tests.ps1` that will check for any git diffs in the `/Docs/*.md` files

## Is there anything particularly tricky?
N/A

## How should this be tested?
1. Create a new Parameter for `Get-JCUser` such as this:
```powershell
[Parameter(ValueFromPipelineByPropertyName, ParameterSetName = 'SearchFilter', HelpMessage = 'Testing parameter for Build-HelpFiles.')]
[String]$RandomAttribute,
```
2. Run the `HelpFiles.Tests.ps1` File
3. It should fail with the following warning:
`WARNING: Diff found in the file /Users/gwein/Documents/GitHub/support/PowerShell/JumpCloud Module/Docs/Get-JCUser.md when we expected none to exist; Please run Build-HelpFiles and commit the results`


[CUT-4128]: https://jumpcloud.atlassian.net/browse/CUT-4128?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
